### ISSUE TITLE
Result_visual3-3.pl ERROR

### ISSUE THREAD

**Issue by junyanzho** [xunchen85/VIcaller#9]
Dear Developer,
I encounter a error while run VIcaller, error message “Modification of non-creatable array value attempted, subscript -1 at Result_visual3-3.pl”

When I looked into the script “Result_visual3-3.pl” , I found that input file “$ARGV[0]_sf.fuq” maybe not exist in any scripts.

Could you help me check it?

Thanks!
JY

**Comment by xunchen85** [comment]
Hi JY,

Did you have "${sample}_sf.fuq" file after your running VIcaller? Can you show me the list of files you currently have?

Best,
Xun

**Comment by junyanzho** [comment]
Hi, xunchen

After run VIcaller, 3 files with suffix 'fuq': ${sample}_1.1fuq, ${sample}_2.1fuq, ${sample}_1sf.fuq.
I try to step by step run, and found the file ${sample}_sf.fuq is not exist.

**Comment by junyanzho** [comment]
Hi, xunchen,
VIcaller generate files as below show(by time):
sample_pe.bam
sample_soft.fastq.gz
sample_h1.sort.bam
sample_h1_1.1fq
sample_h1_2.1fq
sample_h1_h.bam
sample_su.bam
sample_sm.bam
sample_h1_pe.bam
sample_h1_soft.fastq.gz
sample_h1_h1.sort.bam
sample_1sf.fastq
IlluQC_Filtered_files
sample_1.1fq
sample_2.1fq
sample_vsu.sort.bam
sample_vsoft_sort.bam
sample.type
sample_In.virus_v3
sample_1.1fuq
sample_2.1fuq
sample_1sf.fuq
sample_1sf.othu
sample.3
sample.error
sample_f2
sample.virus_f
sample.virus_f2
sample.visualization
sample.fine_mapped
sample_vector.sam
sample_vector.hmap
sample_vector.hunmapped
sample_vector_sf.sam
sample_vector_sf.hmap
sample_vector_sf.hunmapped
sample_vector.hmapped
sample.output
 
Thanks a lot
JY

**Comment by xunchen85** [comment]
Hi JY,

Thanks for pointing it out. The file name in the script should be changed to "1sf.fuq". It will only impact the read visualization but not viral integrations detection.

I have revised the corresponding script.

Thanks,
Xun

**Comment by junyanzho** [comment]
Hi xunchen,

Thank you for your reply and help.
I have another question about virus genome. For HBV integration detection, can I just take HBV genome as virus reference?
Simply extract information of HBV reference NC_003977.1 from files virus_db_090217.fa, virus_db_090217.taxonomy and virus_db_090217.virus_list as a custom reference.

Best,
JY

**Comment by xunchen85** [comment]
Yes if you change the viral database and taxonomy information consistently.

Best,
Xun
